File: Task 1/Java/app/src/main/java/dev/ogabek/java/manager/RoomManager.java
package dev.ogabek.java.manager;

import android.content.Context;

import androidx.room.Database;
import androidx.room.Room;
import androidx.room.RoomDatabase;

import dev.ogabek.java.database.NoteDao;
import dev.ogabek.java.model.Note;

@Database(entities = {Note.class}, version = 1)
public abstract class RoomManager extends RoomDatabase {

    public abstract NoteDao noteDao();
    private static RoomManager INSTANCE;

    public static RoomManager getDatabase(final Context context) {
        if (INSTANCE == null) {
            synchronized (RoomManager.class) {
                if (INSTANCE == null) {
                    INSTANCE = Room.databaseBuilder(context.getApplicationContext(),
                            RoomManager.class, "notes.db")
                            .fallbackToDestructiveMigration()
//                            .allowMainThreadQueries()           // Run in Main Thread
                            .build();
                }
            }
        }
        return INSTANCE;
    }

}

File: Task 1/Kotlin/app/src/main/java/dev/ogabek/kotlin/manager/RoomManager.kt
package dev.ogabek.kotlin.manager

import android.content.Context
import androidx.room.Database
import androidx.room.Room
import androidx.room.RoomDatabase
import dev.ogabek.kotlin.database.NoteDao
import dev.ogabek.kotlin.model.Note

@Database(entities = [Note::class], version = 1)
abstract class RoomManager: RoomDatabase() {

    abstract fun noteDao(): NoteDao

    companion object {

        @Volatile
        private var INSTANCE: RoomManager? = null

        fun getDatabase(context: Context): RoomManager {
            synchronized(this) {
                var instance = INSTANCE
                if (instance == null) {
                    instance = Room.databaseBuilder(context.applicationContext,
                    RoomManager::class.java, "notes.db")
                        .fallbackToDestructiveMigration()
                        .allowMainThreadQueries()       // Run in Main Thread
                        .build()
                    INSTANCE = instance
                }
                return instance
            }
        }

    }

}